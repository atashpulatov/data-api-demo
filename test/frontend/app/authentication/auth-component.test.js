import React from 'react'; // eslint-disable-line
import { shallow } from 'enzyme';
import AuthComponent from '../../../../src/frontend/app/authentication/auth-component.jsx'; // eslint-disable-line no-unused-vars

describe('AuthComponent', () => {
    const location = {};

    beforeAll(() => {
        const origin = { pathname: '/' };
        const state = { origin: origin };
        location.state = state;
    });

    it('should render my component', () => {
        // when
        const component = shallow(<AuthComponent location={location} />);
        // then
        expect(component.getElements()).toMatchSnapshot();
    });
});
