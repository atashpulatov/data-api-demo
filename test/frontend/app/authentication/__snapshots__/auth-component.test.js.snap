// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthComponent should render my component 1`] = `
Array [
  <div>
    <header>
      <h1
        id="authenticate-message"
      >
        Connect to MicroStrategy Environment
      </h1>
    </header>
    <form
      onSubmit={[Function]}
    >
      <div
        className="grid-container padding"
      >
        <label
          className="grid-item grid-item-label"
        >
          Username
        </label>
        <input
          className="grid-item grid-item-input"
          name="username"
          onChange={[Function]}
          type="text"
          value=""
        />
        <label
          className="grid-item grid-item-label"
        >
          Password
        </label>
        <input
          className="grid-item grid-item-input"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
        <label
          className="grid-item grid-item-label"
        >
          Environment URL
        </label>
        <input
          className="grid-item grid-item-input"
          name="envUrl"
          onChange={[Function]}
          type="text"
          value=""
        />
        <input
          className="grid-item-2 button-submit"
          type="submit"
          value="Submit"
        />
      </div>
    </form>
  </div>,
]
`;
